tests/hotplug/cpu.go

```diff
@@ -293,12 +293,11 @@ var _ = Describe("[sig-compute]CPU Hotplug", decorators.SigCompute, decorators.S
 			))
 
 			By("Ensuring the libvirt domain has 4 enabled cores")
-			Eventually(func() error {
+			Eventually(func() cpuCount {
 				domSpec, err = libdomain.GetRunningVMIDomainSpec(vmi)
-				return err
-			}).WithTimeout(20 * time.Second).WithPolling(time.Second).Should(Succeed())
-
-			Expect(countDomCPUs(domSpec)).To(Equal(cpuCount{
+				Expect(err).NotTo(HaveOccurred())
+				return countDomCPUs(domSpec)
+			}).WithTimeout(20 * time.Second).WithPolling(time.Second).Should(Equal(cpuCount{
```

Comment:
I see that in the previous version we retried on error, would you like to preserve this behavior by passing the Gomega parameter in the function signature i.e. `Eventually(func(g Gomega) cpuCount { ` and then use `g.Expect` this way the loop will re-run upon error.

Reply:
https://github.com/kubevirt/kubevirt/pull/14540